sqlmesh/dbt/context.py

```diff
@@ -229,6 +227,9 @@ def jinja_globals(self) -> t.Dict[str, JinjaGlobalAttribute]:
             output["project_name"] = self.project_name
         if self._target is not None:
             output["target"] = self._target.attribute_dict()
+        # pass user-specified default dialect if we have already loaded the config
+        if self.sqlmesh_config.dialect:
+            output["dialect"] = self.sqlmesh_config.dialect
```

Comment:
Since we now need to handle dialect in the jinja, would it make sense to always pass in dialect here? It would simplify `create_builtin_globals` code.

Reply:
Also, do we want to pass in the default dialect? Wouldn't it make more sense to pass in the model dialect during `to_sqlmesh` in the model?

Reply:
We can't pass in self.default_dialect because target hasn't been set the first time this is called and we'll error.

I'm not following your second comment - where in `to_sqlmesh` can we modify the globals? Here?
https://github.com/TobikoData/sqlmesh/blob/b7c659e66b6eab5ec30a6f5d341937d8746de683/sqlmesh/dbt/model.py#L382

Reply:
@crericha  I think this is only relevant in the context of dbt projects and doesn't apply to native projects, so not sure whether we need the dialect as a top-level argument.